### ISSUE TITLE
LayerNormalization crashes on empty inputs when run on CPU

### ISSUE THREAD

**Issue by guillaumekln** [tensorflow/tensorflow#46366]
**System information**
- Have I written custom code: Yes
- OS Platform and Distribution: Ubuntu 18.04
- TensorFlow installed from: binary
- TensorFlow version: 2.4.0
- Python version: 3.6.9

**Describe the current behavior**

`tf.keras.layers.LayerNormalization` crashes when the input is empty and the layer is executed on CPU.

**Describe the expected behavior**

The layer should not crash but return a tensor with the same shape.

**Standalone code to reproduce the issue**

```python
import os
os.environ["CUDA_VISIBLE_DEVICES"] = ""
import tensorflow as tf
layer = tf.keras.layers.LayerNormalization()
layer(tf.zeros([1, 0, 10]))
```

**Other info / logs**

The code above exits with this error:

```text
Floating point exception (core dumped)
```

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with TF v2.3, TF v2.4 and TF-nightly. Colab session crashes on running the code, please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/a17885e619f64587f0946ef115003ff9/46366.ipynb#scrollTo=L8qDirj_Zx4v). Thanks!

**Comment by around-star** [comment]
@guillaumekln , @amahendrakar , @jvishnuvardhan a tensor of shape [1,0,10] would return a tensor
 `tf.Tensor([], shape=(1, 0, 10), dtype=float32) TensorShape([1, 0, 10])` , Any tensor of value [] would cause the runtime to crash. I will send a pull request to raise a valid error regarding the faulty input tensor.

**Comment by bhack** [comment]
Can we close this?

**Comment by guillaumekln** [comment]
The crash is not fixed. I believe the `FusedBatchNorm` CPU kernel should check if the input is empty. I tried to fix the issue but eventually moved to something else.

**Comment by bhack** [comment]
The PRs at the python level was rejected. Do you think that will be accepted at cpp level?

**Comment by guillaumekln** [comment]
The GPU kernel [is checking for empty inputs](https://github.com/tensorflow/tensorflow/blob/v2.5.0-rc1/tensorflow/core/kernels/fused_batch_norm_op.cc#L818), but not the CPU kernel. So I think a code change would make sense here.

**Comment by bhack** [comment]
@nikitamaia I think that we could remove the keras label here as this is a contribution at c++ level.

**Comment by guillaumekln** [comment]
Yes, the issue is more specifically to `tf.compat.v1.nn.fused_batch_norm` that is called by `tf.keras.layers.LayerNormalization`.

**Comment by saikumarchalla** [comment]
Still an issue in TF 2.6 Nightly as well.Thanks!

**Comment by oawxkw** [comment]
I also observed the following API aliases can cause the same issue in older versions of tensorflow.
Users should be cautious when using them on CPU up to tensorflow 2.5.1 (v2.5.0-160-g8222c1cfc86).

- `(tf.keras.layers.LayerNormalization)`, `tf.compat.v1.keras.layers.LayerNormalization`

<details>
  <summary>Code to reproduce the issue in <code>tf.compat.v1.keras.layers.LayerNormalization</code> in older versions</summary>

```python
import tensorflow as tf
print(tf.version.GIT_VERSION, tf.version.VERSION, flush=True)
print(tf.config.list_physical_devices(), flush=True)


try:
    layer = tf.compat.v1.keras.layers.LayerNormalization()
    layer(tf.zeros([1, 0, 10]))
except Exception as e:
    print("Error:", str(e), flush=True)
print("Success!", flush=True)
```

On my CPU machine, the process aborts with a Floating point exception(core dumped), which is not expected.

```text
v2.5.0-160-g8222c1cfc86 2.5.1
[PhysicalDevice(name='/physical_device:CPU:0', device_type='CPU')]
Floating point exception(core dumped)
```
</details>

It seems to be fixed in tensorflow 2.6.0 (v2.6.0-rc2-32-g919f693420e) and later versions.
